File: snippets/fsharp/System/Decimal/.ctor/ctorl.fs
module ctorl

//<Snippet3>
// Example of the decimal( int64 ) constructor.
open System

// Create a decimal object and display its value.
let createDecimal (value: int64) valToStr =
    let decimalNum = Decimal value

    // Format the constructor for display.
    let ctor = $"decimal( {valToStr} )"

    // Display the constructor and its value.
    printfn $"{ctor,-35}{decimalNum,22}"

printfn "This example of the decimal( int64 ) constructor\ngenerates the following output.\n"
printfn "%-35s%22s" "Constructor" "Value"
printfn "%-35s%22s" "-----------" "-----"

// Construct decimal objects from long values.
createDecimal Int64.MinValue "Int64.MinValue"
createDecimal Int64.MaxValue "Int64.MaxValue"
createDecimal 0L "0L"
createDecimal 999999999999999999L "999999999999999999L"
createDecimal 0x2000000000000000L "0x2000000000000000L"
createDecimal (int64 0xE000000000000000L) "int64 0xE000000000000000L"

// This example of the decimal( int64 ) constructor 
// generates the following output.
//
// Constructor                                         Value
// -----------                                         -----
// decimal( Int64.MinValue )            -9223372036854775808
// decimal( Int64.MaxValue )             9223372036854775807
// decimal( 0L )                                           0
// decimal( 999999999999999999L )         999999999999999999
// decimal( 0x2000000000000000L )        2305843009213693952
// decimal( int64 0xE000000000000000L )  -2305843009213693952
//</Snippet3> 

File: snippets/csharp/System/TimeSpan/.ctor/ctorl.cs
//<Snippet1>
// Example of the TimeSpan( long ) constructor.
using System;

class TimeSpanCtorLDemo
{
    // Create a TimeSpan object and display its value.
    static void CreateTimeSpan(long ticks)
    {
        TimeSpan elapsedTime = new(ticks);

        // Format the constructor for display.
        string ctor = $"TimeSpan( {ticks} )";

        // Pad the end of a TimeSpan string with spaces if
        // it does not contain milliseconds.
        string elapsedStr = elapsedTime.ToString();
        int pointIndex = elapsedStr.IndexOf(':');

        pointIndex = elapsedStr.IndexOf('.', pointIndex);
        if (pointIndex < 0) elapsedStr += "        ";

        // Display the constructor and its value.
        Console.WriteLine($"{ctor,-33}{elapsedStr,24}");
    }

    static void Main()
    {
        Console.WriteLine(
            "This example of the TimeSpan( long ) constructor " +
            "\ngenerates the following output.\n");
        Console.WriteLine($"{"Constructor",-33}{"Value",16}");
        Console.WriteLine($"{"-----------",-33}{"-----",16}");

        CreateTimeSpan(1);
        CreateTimeSpan(999999);
        CreateTimeSpan(-1000000000000);
        CreateTimeSpan(18012202000000);
        CreateTimeSpan(999999999999999999);
        CreateTimeSpan(1000000000000000000);
    }
}

/*
This example of the TimeSpan( long ) constructor
generates the following output.

Constructor                                 Value
-----------                                 -----
TimeSpan( 1 )                            00:00:00.0000001
TimeSpan( 999999 )                       00:00:00.0999999
TimeSpan( -1000000000000 )            -1.03:46:40
TimeSpan( 18012202000000 )            20.20:20:20.2000000
TimeSpan( 999999999999999999 )   1157407.09:46:39.9999999
TimeSpan( 1000000000000000000 )  1157407.09:46:40
*/
//</Snippet1>
